Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -166,8 +166,9 @@ public struct SwiftlyHTTPClient {
         for try await buffer in response.body {
             receivedBytes += buffer.readableBytes
 
-            try buffer.withUnsafeReadableBytes { bufferPtr in
-                try fileHandle.write(contentsOf: bufferPtr)
+            let byteData = buffer.getData(at: buffer.readerIndex, length: buffer.readableBytes)
+            if let data = byteData {
+                try fileHandle.write(contentsOf: data)
```

Comment:
Can probably replace all of this with 
```swift
try fileHandle.write(contentsOf: buffer.readableBytesView)
```

Reply:
Thank you for the suggestion, this is much cleaner. I've pushed an update that uses it.